src/analyzer.rs

```diff
@@ -182,6 +182,44 @@ impl ImportAssertions {
       _ => None,
     }
   }
+
+  pub fn is_known_none(&self, key: &str) -> bool {
+    match self {
+      ImportAssertions::None => true,
+      ImportAssertions::Known(map) => !map.contains_key(key),
+      _ => false,
+    }
+  }
+
+  pub fn is_empty(&self) -> bool {
+    matches!(self, ImportAssertions::None)
+  }
+
+  /// Try to get a flat dictionary, successful if there are no unknowns.
+  pub fn maybe_static(&self) -> Option<HashMap<String, String>> {
```

Comment:
Where is this used? Is it necessary? We should minimize having methods on the that allocate while analyzing a graph. For example, maybe this should instead be a method that verifies the `ImportAssertions` then someone could use `.get` afterwards instead or this should return a new struct where this method verifies the `ImportAssertions` then has a reference to the old map while providing some `.get` methods.

Reply:
Ah I was going to use this as an alternative serialization for static imports but changed my mind (deferred for the future?), I'll remove this method.